staging/src/k8s.io/client-go/tools/leaderelection/example/main.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
+	"k8s.io/klog"
+)
+
+// main demonstrates a leader elected process that will step down if interrupted.
+func main() {
+	klog.InitFlags(nil)
+	flag.Parse()
+	args := flag.Args()
+	if len(args) != 3 {
+		log.Fatalf("requires three arguments: ID NAMESPACE CONFIG_MAP_NAME (%d)", len(args))
+	}
+
+	// leader election uses the Kubernetes API by writing to a ConfigMap or Endpoints
+	// object. Conflicting writes are detected and each client handles those actions
+	// independently.
+	var config *rest.Config
+	var err error
+	if kubeconfig := os.Getenv("KUBECONFIG"); len(kubeconfig) > 0 {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	} else {
+		config, err = rest.InClusterConfig()
+	}
+	if err != nil {
+		log.Fatalf("failed to create client: %v", err)
+	}
+
+	// we use the ConfigMap lock type since edits to ConfigMaps are less common
+	// and fewer objects in the cluster watch "all ConfigMaps" (unlike the older
+	// Endpoints lock type, where quite a few system agents like the kube-proxy
+	// and ingress controllers must watch endpoints).
+	id := args[0]
+	lock := &resourcelock.ConfigMapLock{
+		ConfigMapMeta: metav1.ObjectMeta{
+			Namespace: args[1],
+			Name:      args[2],
+		},
+		Client: kubernetes.NewForConfigOrDie(config).CoreV1(),
+		LockConfig: resourcelock.ResourceLockConfig{
+			Identity: id,
```

Comment:
Doesn't this need to be unique per instance of this main?

Reply:
Yes, as a test example you have to pass that in as an argument.